scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -1,6 +1,7 @@
 import warnings
 import numpy as np
-from numpy.testing import assert_array_almost_equal, assert_array_equal
+from numpy.testing import (assert_almost_equal, assert_array_almost_equal,
```

Comment:
Should I also replace existing `assert_almost_equal` in the other tests in the file too?

Reply:
@windows-server-2003, you don't have to replace existing occurrences of those test functions in code that you are not contributing.  Just be sure to use the recommended ones in new code.

This is a frequently occurring issue in pull requests.  I think we (the SciPy devs) should consider doing a bulk replacement of all the discommended (i.e. no longer recommended but not actually deprecated) functions, to avoid the natural tendency of contributors to copy the existing tests.

Reply:
Thanks, understood.

Reply:
> should we perhaps prefer `assert_allclose` to `assert_almost_equal` in new test code? 

Yes, we decided that some time ago, and made note of that decision in the "missing bits" developer documentation: https://docs.scipy.org/doc/scipy/dev/missing-bits.html#test-functions-from-numpy-testing

@windows-server-2003, we don't expect all new contributors to be familiar with that document, but now that you have a couple PRs in progress, it would be worthwhile to browse through it.